### PR TITLE
S2 hiring partner send ajax

### PR DESCRIPTION
**Pull request by imogenhallett**
Created test form with required field and postcode validation with test controller.

Test files are:
/templates/test.phtml
/src/routes.php $app->post('/api/addHiringPartner'

To be reviewed:
/public/js/addHiringPartner.js

### CODE REVIEW

File: public/js/addHiringPartner.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+/**
+ * Calls the ajax request and displays response/error message
+ *
+ * @param formData {object} values from form
+ */
+function sendRequest(formData) {
+    fetchData(formData).then(function (data) {
+        console.log(data)
```
Comment (felixgate13) [inline public/js/addHiringPartner.js]: cheeky console.log here

Comment (imogenhallett) [reply public/js/addHiringPartner.js]: Fixed in c84ec38

**Comment by imogenhallett** [comment]
Since last review we have linked form from main branch to use the ajax.

Files to be reviewed:
js/addHiringPartner.js
templates/addHiringPartner.phtml
src/routes.php

File: src/routes.php [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+use \Slim\Http\Response as Response;
 
 //FrontEnd
 $app->get('/','HomePageController');
 $app->get('/admin', 'AdminController');
 $app->get('/register', 'RegisterController');
-$app->get('/addPartner', function ($request, $response, $args) {
+$app->get('/addHiringPartner', function ($request, $response, $args) {
     return $this->renderer->render($response, 'addHiringPartner.phtml', $args);
 });
```
Comment (hoptond) [inline src/routes.php]: The link on the admin page refers to /addPartner and not /addHiringPartner. Trying to navigate here via results in Page Not Found.

Comment (hoptond) [inline src/routes.php]: Is there a reason you didn't do this in a dedicated controller?

Comment (hoptond) [inline src/routes.php]: On that note, I can straight up get to this page without having to log in.

File: templates/addHiringPartner.phtml [hunk 3]
```diff
@@ -1,42 +1,52 @@
...
+            <input type="text" id="postcode" class="addHiringPartner field" name="postcode" data-required="true" data-postcode="true" required>
+        </div>
+        <div class="formField">
             <label for="telNumber" class="hiringPartnerLabels">Phone Number</label>
-            <input type="tel" id="telNumber" class="addHiringPartner" name="telNumber">
+            <input type="tel" id="telNumber" class="addHiringPartner field" name="telNumber">
+        </div>
+        <div class="formField">
             <label for="companyURL" class="hiringPartnerLabels">Website</label>
-            <input type="url" id="companyURL" class="addHiringPartner" name="companyURL">
```
Comment (hoptond) [inline templates/addHiringPartner.phtml]: This url nags you for a http:// or https:// prefix. Might be a bit annoying if you're straight up typing an address without it.

File: templates/addHiringPartner.phtml [hunk 4]
```diff
@@ -1,42 +1,52 @@
...
+    <div>
+        <a href="/admin" class="returnButton">Return to admin page</a>
+    </div>
+    <h1>Add Hiring Partner Page</h1>
+    <div id="message"></div>
+    <form id="addHiringPartnerForm">
+        <div class="formField">
             <label for="companyName" class="hiringPartnerLabels">Company name</label>
-            <input type="text" id="companyName" class="addHiringPartner" name="companyName" required>
+            <input type="text" id="companyName" class="addHiringPartner field" name="companyName" data-required="true" required>
```
Comment (hoptond) [inline templates/addHiringPartner.phtml]: I can submit a blank company name by submitting '  '. Same for the tech stack.

File: src/routes.php (hunk 2 above)
Comment (imogenhallett) [reply src/routes.php]: Fixed in 82df36e

Comment (imogenhallett) [reply src/routes.php]: This was done for testing.  Setting up the controller has been done in another task.

Comment (imogenhallett) [reply src/routes.php]: Fixed in 083715b

File: templates/addHiringPartner.phtml (hunk 3 above)
Comment (imogenhallett) [reply templates/addHiringPartner.phtml]: This is how `type="url" ` works I'm afraid

File: templates/addHiringPartner.phtml (hunk 4 above)
Comment (imogenhallett) [reply templates/addHiringPartner.phtml]: This validation is to stop people forgetting to input values rather than enforcing correct values.

**Pull request merged**